Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XLGToDBAnalyzer.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using BuildXL.Scheduler.Tracing;
+using BuildXL.ToolSupport;
+using BuildXL.Utilities;
+using Google.Protobuf;
+using BuildXL.Engine.Cache;
+using BuildXL.Engine.Cache.KeyValueStores;
+using BuildXL.Engine.Cache.Serialization;
+using BuildXL.Native.IO;
+using System.Diagnostics;
+using BuildXL.Execution.Analyzer.Model;
+using BuildXL.Analyzers.Core.XLGPlusPlus;
+
+namespace BuildXL.Execution.Analyzer
+{
+    internal partial class Args
+    {
+        public Analyzer InitializeXLGToDBAnalyzer()
+        {
+            string outputDirPath = null;
+            foreach (var opt in AnalyzerOptions)
+            {
+                if (opt.Name.Equals("outputDir", StringComparison.OrdinalIgnoreCase) ||
+                   opt.Name.Equals("o", StringComparison.OrdinalIgnoreCase))
+                {
+                    outputDirPath = ParseSingletonPathOption(opt, outputDirPath);
+                }
+                else
+                {
+                    throw Error("Unknown option for event stats analysis: {0}", opt.Name);
+                }
+            }
+
+
+            if (string.IsNullOrEmpty(outputDirPath))
+            {
+                throw Error("/outputDir parameter is required");
+            }
+
+
+            return new XLGToDBAnalyzer(GetAnalysisInput())
+            {
+                OutputDirPath = outputDirPath,
+            };
+        }
+
+        /// <summary>
+        /// Write the help message when the analyzer is invoked with the /help flag
+        /// </summary>
+        private static void WriteXLGToDBHelp(HelpWriter writer)
+        {
+            writer.WriteBanner("XLG to DB \"Analyzer\"");
+            writer.WriteModeOption(nameof(AnalysisMode.XlgToDb), "Dumps event data from the xlg into a database.");
+            writer.WriteOption("outputDir", "Required. The directory to write out the RocksDB database", shortName: "o");
+        }
+    }
+
+
+    /// <summary>
+    /// Analyzer to dump xlg events and other data into RocksDB
+    /// </summary>
+    internal sealed class XLGToDBAnalyzer : Analyzer
+    {
+        public string OutputDirPath;
+        private bool m_accessorSucceeded = true;

```

Comment:
>private bool m_accessorSucceeded = true; [](start = 8, length = 40)

the default value does not agree with `Accessor` default value - you should be setting this to true only when the accessor is set